### PR TITLE
April 2023

### PR DESCRIPTION
**Pull request by SethBarberee**
Mix of documentation, decomp, and finally wiring up section generation.

Was wondering why they didn't work until I read the documentation for gcc :smiling_face_with_tear: .

### CODE REVIEW

File: src/code_800B5F0.c [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    *(vu16 *)BG_PLTT = RGB_WHITE;
+
+    ie_store = REG_IE;
+    REG_IE = INTR_FLAG_KEYPAD | INTR_FLAG_GAMEPAK;
+
+    REG_KEYCNT = KEY_AND_INTR | R_BUTTON | L_BUTTON | SELECT_BUTTON; // 0x8304
+    REG_IME = 1;
+
+    SoundBiasReset();
+    asm("swi 0x3");
```
Comment (AnonymousRandomPerson) [inline src/code_800B5F0.c:294]: Add NONMATCHING.

File: src/code_8045A00.c [hunk 2]
```diff
@@ -128,3 +132,69 @@ void sub_8045C28(struct Item *Item, u8 itemID, u32 param_3)
...
+          shopFlag = TRUE;
+          uVar5 = 1;
+        }
+        else
+        {
+          if ((tile->terrainType & (TERRAIN_TYPE_NORMAL | TERRAIN_TYPE_SECONDARY)) == 0) {
+            uVar5 = 3;
+          }
+          else {
+            // TODO: Ternary?
```
Comment (AnonymousRandomPerson) [inline src/code_8045A00.c:184]: Is this TODO still needed?

File: src/options_menu.c [hunk 3]
```diff
@@ -103,34 +103,44 @@ enum
     MENU_OPTION_NO
 };
 
+enum OptionsMenuStates {
+    OPTIONS_MENU_INIT,
+    OPTIONS_MENU_MAIN,
+    OPTIONS_MENU_EXIT,
+    OPTIONS_MENU_PRE_HINT_SELECTION,
+    OPTIONS_MENU_HINT_SELECTION,
+    OPTIONS_MENU_DISPLAY_HINT,
```
Comment (AnonymousRandomPerson) [inline src/options_menu.c:112]: The unknown enum for 6 can be defined as something like `OPTIONS_MENU_UNKNOWN` rather than skipping it.

File: src/weather.c [hunk 4]
```diff
@@ -1,15 +1,196 @@
...
+
+void sub_807E254(struct Entity *pokemon,struct Entity *target)
+{
+  struct Position pokemonPos;
+  struct Position targetPos;
+  
+
+  if (HasAbility(pokemon, ABILITY_SUCTION_CUPS)) {
+    SetMessageArgument(gAvailablePokemonNames,pokemon,0);
+    sub_80522F4(pokemon,target,*gUnknown_80FCAC0); // $m0 is anchored! It won't switch places
```
Comment (AnonymousRandomPerson) [inline src/weather.c:91]: The functions added to this file are not related to weather. Let's move them to a new file.

File: src/code_800B5F0.c (hunk 1 above)
Comment (SethBarberee) [reply src/code_800B5F0.c:294]: Not really a NONMATCHING as this is an intentional asm call to call the interrupt. There isn't a C equivalent for this. We do a similar thing with `Hang`.

Comment (AnonymousRandomPerson) [reply src/code_800B5F0.c:294]: Gotcha, carry on then.

File: src/code_8045A00.c (hunk 2 above)
Comment (SethBarberee) [reply src/code_8045A00.c]: Removed

File: src/options_menu.c (hunk 3 above)
Comment (SethBarberee) [reply src/options_menu.c:112]: Added `OPTIONS_MENU_UNKNOWN_6`

File: src/weather.c (hunk 4 above)
Comment (SethBarberee) [reply src/weather.c]: Moved

**Review by AnonymousRandomPerson — APPROVED** [review]

File: src/code_800B5F0.c (hunk 1 above)
Comment (Kermalis) [reply src/code_800B5F0.c:294]: Is there a way to have an inline function that calls these SWI? Mario & Luigi does that.
